backends/dpdk/dpdkAsmOpt.h

```diff
@@ -135,6 +135,27 @@ class RemoveLabelAfterLabel : public Transform {
     }
 };
 
+
+// This pass Collects all metadata struct member used in program
+class CollectUsedMetadataField : public Inspector {
+    ordered_set<cstring> used_fields;
+  public:
+    bool preorder(const IR::Member *m) override {
```

Comment:
This is not necessarily a field of the metadata structure, but I guess it will include all the fields. So you may keep some extra fields here if they happen to show up in other structures.

Reply:
Now only insert if it's metadata's member.